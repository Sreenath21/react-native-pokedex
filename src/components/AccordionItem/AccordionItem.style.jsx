import {StyleSheet} from 'react-native';
import colors from '../../utils/colors';

export default StyleSheet.create({
  container: {
    borderColor: colors.primary,
    borderWidth: 1,
    borderRadius: 5,
    marginBottom: 20,
  },
  header: {
    flexDirection: 'row',
    alignItems: 'center',
    paddingHorizontal: 20,
    paddingVertical: 10,
  },
  title: {
    flex: 1,
    fontSize: 18,
    fontWeight: 'bold',
    borderRightColor: '#ddd',
    borderRightWidth: 1,
    color: colors.primary,
  },
  status: {
    flex: 2,
    fontSize: 18,
    paddingLeft: 10,
    color: colors.primary,
  },
  expandedBtn: {
    paddingHorizontal: 8,
    paddingVertical: 2,
    borderRadius: 30,
    borderWidth: 1,
    borderColor: colors.primary,
    alignItems: 'center',
    justifyContent: 'center',
  },
  expandedBtnText: {
    color: colors.primary,
    fontSize: 18,
  },
  body: {
    borderTopWidth: 1,
    borderTopColor: '#ddd',
    marginHorizontal: 20,
    paddingTop: 20,
    paddingBottom: 10,
  },
});
