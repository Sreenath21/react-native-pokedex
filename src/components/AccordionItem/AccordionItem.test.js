import React from 'react';
import {View, Text} from 'react-native';
import AccordionItem from './AccordionItem';
import renderer from 'react-test-renderer';

it('Checking AccordionItem rendering', () => {
  const tree = renderer.create(
    <AccordionItem
      title="Types"
      children={
        <View>
          <Text>Normal</Text>
        </View>
      }
    />,
  );

  expect(tree).toMatchSnapshot();
});
