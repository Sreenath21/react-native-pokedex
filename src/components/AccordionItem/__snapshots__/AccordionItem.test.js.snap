// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checking AccordionItem rendering 1`] = `
<View
  style={
    {
      "borderColor": "#391363",
      "borderRadius": 5,
      "borderWidth": 1,
      "marginBottom": 20,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "paddingHorizontal": 20,
        "paddingVertical": 10,
      }
    }
  >
    <Text
      style={
        {
          "borderRightColor": "#ddd",
          "borderRightWidth": 1,
          "color": "#391363",
          "flex": 1,
          "fontSize": 18,
          "fontWeight": "bold",
        }
      }
    >
      Types
    </Text>
    <Text
      style={
        {
          "color": "#391363",
          "flex": 2,
          "fontSize": 18,
          "paddingLeft": 10,
        }
      }
    >
      Status
    </Text>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "borderColor": "#391363",
          "borderRadius": 30,
          "borderWidth": 1,
          "justifyContent": "center",
          "paddingHorizontal": 8,
          "paddingVertical": 2,
        }
      }
    >
      <Text
        style={
          {
            "color": "#391363",
            "fontSize": 18,
          }
        }
      >
        +
      </Text>
    </View>
  </View>
</View>
`;
